pybandits/model.py

```
@@ -1212,20 +1229,46 @@ def _prepare_context_arrays(self, context: np.ndarray) -> Tuple[np.ndarray, Dict
+        if lr_scheduler_type is not None:
+            if lr_scheduler_type not in self._supported_lr_schedulers:
+                raise ValueError(
+                    f"Invalid lr_scheduler_type: {lr_scheduler_type}. "
+                    f"Supported schedulers are: {list(self._supported_lr_schedulers.keys())}"
+                )
+            try:
+                learning_rate = self._supported_lr_schedulers[lr_scheduler_type](
+                    init_value=learning_rate, **lr_scheduler_kwargs
+                )
+            except (TypeError, ValueError) as e:
+                raise e.__class__(f"Invalid lr_scheduler_kwargs: {lr_scheduler_kwargs}.\n{e}")
```

Ruff (0.15.9)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)